cmd/create/cluster/cmd.go

```diff
@@ -1917,6 +1943,113 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	isSharedVPC := false
+	privateHostedZoneID := strings.Trim(args.privateHostedZoneID, " \t")
+	privateHostedZoneRoleARN := strings.Trim(args.privateHostedZoneRoleARN, " \t")
+	baseDomain := strings.Trim(args.baseDomain, " \t")
+	if privateHostedZoneID != "" ||
+		privateHostedZoneRoleARN != "" ||
+		baseDomain != "" {
+		if !isSTS {
+			r.Reporter.Errorf("Installing a cluster into a shared VPC is only supported for STS clusters")
+			os.Exit(1)
+		}
+		if len(subnetIDs) == 0 {
+			r.Reporter.Errorf("Installing a cluster into a shared VPC is only supported for BYO VPC clusters")
+			os.Exit(1)
+		}
+		if isHostedCP {
```

Comment:
Same here, we have a check in the backend, I assume at a certain point it will be supported for HCP. I think it's ok to remove the validation, as the error will be returned from the backend.

Reply:
removed the check